src/plugins/vis_type_vislib/public/line_to_expression.ts

```diff
@@ -26,9 +27,13 @@ export const toExpressionAst = async (vis: Vis, params: any) => {
   );
 
   // Checks if there are vislayers to overlay. If not, default to the vislib implementation.
-  if (params.visLayers == null || Object.keys(params.visLayers).length === 0) {
+  const dimensions: VislibDimensions = await buildVislibDimensions(vis, params);
+  if (
+    !isEligibleForVisLayers(vis, dimensions) ||
+    params.visLayers == null ||
+    Object.keys(params.visLayers).length === 0
+  ) {
     // This wont work but is needed so then it will default to the original vis lib renderer
```

Comment:
nit: is this comment outdated? not sure what it means exactly

Reply:
To automatically default back to the non-expression method (render using vega-lite), the `toExpressionAST` function needs to fail since in the line definition, we cannot logically choose one over the other.

Reply:
I will create a github issue to look more into this to see what needs to change for it to have a better switching mechanism.

Reply:
It's not failing here though, rather it's just building the rest of the expression string without constructing it via the vega expression functions. I think we can just remove this comment to prevent confusion.

And overall, I think it's actually good design to have the entire string pipeline constructed in here (both vislib and vega implementations). Ideally, all vis types have an implemented `toExpressionAst`

Reply:
Updated the comment. I checked again and it doesn't fail anymore and properly builds the vislib line chart